components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,463 @@
+	local typeStruct = Json.parseIfString(args.type)
+
+	PrizePool._assertOpponentStructType(typeStruct)
+
+	parsedArgs.type = typeStruct.type
+
+	return parsedArgs
+end
+
+function PrizePool:create()
+	self.options = self:_readConfig(self.args)
+	self.prizes = self:_readPrizes(self.args)
+	self.placements = self:_readPlacements(self.args)
+
+	if self:_hasUsdPrizePool() then
+		self:setConfig('showUSD', true)
+		self:addPrize('USD', 1)
+	end
+
+	return self
+end
+
+function PrizePool:build()
+	-- TODO
+	mw.logObject(self)
+	return ''
+end
+
+function PrizePool:_readConfig(args)
+	for name, configData in pairs(config) do
+		local value = configData.default
+		if configData.read then
+			value = Logic.nilOr(configData.read(args), value)
+		end
+		self:setConfig(name, value)
+	end
+
+	return self.options
+end
+
+--- Parse the input for available prize types overall.
+function PrizePool:_readPrizes(args)
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		assert(prizeDatum, 'Error: Missmatch between prizeData and prizeTypes. Is a prizeType missing in prizeData?')
+		local fieldName = prizeDatum.header
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, prizeValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				local data = prizeDatum.headerParse(self, prizeValue, args, index)
+				self:addPrize(prizeEnum, index, data)
+			end
+		end
+	end
+
+	return self.prizes
+end
+
+function PrizePool:_readPlacements(args)
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		if not args[placementIndex] then
+			break
+		end
+
+		local placementInput = Json.parseIfString(args[placementIndex])
+		local placement = Placement(placementInput, currentPlace + 1, self.opponentType)
+
+		currentPlace = placement.placeEnd
+		table.insert(self.placements, placement)
+	end
+
+	return self.placements
+end
+
+function PrizePool:setConfig(option, value)
+	self.options[option] = value
+	return self
+end
+
+function PrizePool:addCustomConfig(name, default, func)
+	config[name] = {
+		default = default,
+		func = func
+	}
+	return self
+end
+
+-- TODO extract parts of this to the future Enum class
+local CUSTOM_TYPES_OFFSET = 100
+local CUSTOM_TYPES_USED = 0
+function PrizePool:addCustomPrizeType(enum, data)
+	prizeTypes[enum] = CUSTOM_TYPES_OFFSET + CUSTOM_TYPES_USED
+	CUSTOM_TYPES_USED = CUSTOM_TYPES_USED + 1
+	prizeData[prizeTypes[enum]] = data
+	return self
+end
+
+function PrizePool:addPrize(enum, index, data)
+	assert(prizeTypes[enum], 'addPrize: Not a valid prizeEnum!')
+	assert(Logic.isNumeric(index), 'addPrize: Index is not numeric!')
+	table.insert(self.prizes, {id = enum .. index, enum = prizeTypes[enum], index = index, data = data})
+	return self
+end
+
+function PrizePool:setWidgetInjector(widgetInjector)
+	assert(widgetInjector:is_a(WidgetInjector), "setWidgetInjector: Not a WidgetIjector")
+	self.widgetInjector = widgetInjector
+	return self
+end
```

Comment:
can we really not come up with something better than `setFooInjector`?

Reply:
Pretty standard setter function naming scheme (set`fieldName`). If you have a better name suggestion, I'm all ears.